qcore/validate_vm.py

```diff
@@ -164,24 +174,60 @@ def vmfile(filename):
     # 9: Check SRF is within bounds of the VM if it is given
     if srf is not None:
         srf_bounds = get_bounds(srf)
-        polygon.append(polygon[0])
-        path = mpltPath.Path(polygon)
+        edges = []
+        for index, start_point in enumerate(polygon):
+            end_point = polygon[(index + 1) % len(polygon)]
+            lons = np.linspace(start_point[0], end_point[0], 10000)
+            lats = compute_intermediate_lat(start_point, end_point, lons)
+            edges.extend(list(zip(lons, lats)))
+
+        path = mpltPath.Path(edges)
         for bounds in srf_bounds:
             if not all(path.contains_points(bounds)):
                 return False, "Srf extents not contained within velocity model corners"
     return True, "VM seems alright: {}.".format(vm_dir)
 
 
+def compute_intermediate_lat(lon_lat1, lon_lat2, lon_in):
+    conversion_factor = np.pi / 180
```

Comment:
Comment about what this function does as it's not intuitive. Maybe a rename might help too?